packages/extension-base/src/koni/api/nft/unique_network_nft/index.ts

```diff
@@ -0,0 +1,164 @@
+    else if (['quartz'].includes(this.chain))
+        {
+            endpoint = QUARTZ_SCAN_ENDPOINT;
+            unique_address = encodeAddress(decodeAddress(address), 255);
+            // Quartz prefix: 255
+        }
+    else if (['opal'].includes(this.chain))
+        {
+            endpoint = OPAL_SCAN_ENDPOINT;
+            unique_address = address;
+            // Opal address: Normal address
+        }
+
+    const resp = await fetch(endpoint, {
+        method: 'post',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(UniqueNftApi.parseNftRequest(unique_address))
+    });
+    
+    const result = await resp.json() as Record<string, any>;
+    return result?.data?.tokens?.data as NftData[];
+  }
+
+  public async handleNfts(params: HandleNftParams) {
+    try {
+      await Promise.all(this.addresses.map(async (address) => {
+        const nfts = await this.getNftByAccount(address);
+
+        if (nfts) {
+          for (const nft of nfts) {
+
+            // Handle properties
+            const propertiesMap: Record<string, any> = {};
+            const att_record = nft.attributes
+            if (att_record) {
+                for (const item in att_record) {
+                    // console.log('sos',att_record[item].value);
+                    if (att_record[item].name._.toLowerCase() == 'traits') {
+                        const traits: string[] = [];
+
+                        const trait_values = att_record[item].value;
+                        for (const trait in trait_values) {
+                            // console.log('trait: ', trait_values[trait]._);
+                            traits.push(trait_values[trait]._);
+                        //     // propertiesMap['traits'] = propertiesMap['traits'].concat(trait_values[trait]._);
+                        };
+                            propertiesMap['traits'] = {
+                                value: traits
+                            };
+                        }
+                    else {
+                        // console.log(a tt_record[item].name._, att_record[item].value._)
+                        propertiesMap[att_record[item].name._] = {
+                            value: att_record[item].value._
+                            };
+                    }
+                }
+            }
+
+            // console.log('sos', propertiesMap);
+
+            const parsedNft: NftItem = {
+              id: nft.token_id.toString(),
+              chain: this.chain,
+              owner: address,
+              name: nft.token_name,
+              image: this.parseUrl(nft.image?.fullUrl),
+              description: nft.collection_description,
+              collectionId: nft.collection_id.toString(),
+            //   properties: nft.attributes as Record<any, any>
+              properties: propertiesMap as Record<any, any>
+            };
+            
+            const parsedCollection: NftCollection = {
```

Comment:
Only update collection info once, avoid the same information being updated multiple times

Reply:
Can you point out where the collection information updates multiple times?

Reply:
- Break NFT properties handling to a new function.
- Optimize to avoid update collection many time.